### PR TITLE
[Live Site] Add backend API calls

### LINKED ISSUE #491: [Live-Site] Add the Join functionality in live share
### AC
- Guest and Host  users will be able to join the live share just by putting their name
- The role of the user will be determined from the query param (role)
- Integrate 100ms SDK to join the dummy room which will be replaced with backed API in future

### UX
https://www.figma.com/file/2VtPGJH61Qp0iarkKW6KTO/Live-Site-Detailed-Designs?type=design&node-id=0-1&t=aWR2yCHfQdp1BaF4-0

### PR DESCRIPTION
**Pull request by SanketDhabarde**
### What is the change?
Previously we were joining the live screen share with the 100ms API. Now we are using our own Backend APIs to create and enter the room

### Is Development Tested?

- [x] Yes
- [ ] No

### Video

https://github.com/Real-Dev-Squad/website-www/assets/40385118/9114a622-f2d2-4b14-8004-9a1c3772cb78

### CODE REVIEW

**Review by satyam73 — CHANGES_REQUESTED** [review]
Also @SanketDhabarde for guest we will use HLS stream right?

**Review by vvaibhavdesai — APPROVED** [review]

File: app/constants/live.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
-export const ROLES = { guest: 'guest', host: 'host', presenter: 'presenter' };
+export const ROLES = {
+  guest: 'guest',
+  host: 'host',
+  presenter: 'presenter',
```
[the remark this answers is not in the document]
Comment (SanketDhabarde) [reply app/constants/live.js:4]: I will remove it. Added during initial roles

File: app/controllers/live.js [hunk 2]
```diff
@@ -28,7 +28,7 @@ export default class LiveController extends Controller {
     super(...arguments);
     setTimeout(() => {
```
[the remark this answers is not in the document]
Comment (SanketDhabarde) [reply app/controllers/live.js:29]: Sure

**Comment by SanketDhabarde** [comment]
> Also @SanketDhabarde for guests we will use HLS stream right?

Yes, that will be when we allow multiple presenters and guests. Currently, the implementation is with one host and numerous guests.

File: app/services/live.js [hunk 3]
```diff
@@ -48,9 +60,56 @@ export default class LiveService extends Service {
...
+      console.error(error);
+    }
+  }
+
+  async joinSession(userName, role, room) {
     try {
-      const token = await this.getToken(userName, userType);
+      const roomId =
+        ROLES.host === role ? await this.createRoom(userName) : room;
+      console.log({ roomId }); // For now use it to create link for guest
```
Comment (harshith-venkatesh) [inline app/services/live.js:110]: Why are we logging roomId here?
Is there a significance for the same

File: tests/acceptance/live-test.js [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
-import { visit, currentURL, waitFor, typeIn, click } from '@ember/test-helpers';
+import { visit, currentURL, typeIn, click } from '@ember/test-helpers';
 import { setupApplicationTest } from 'website-www/tests/helpers';
 
 module('Acceptance | live', function (hooks) {
   setupApplicationTest(hooks);
   // TODO - remove test for dev part when it goes to production
   test('skeleton and video should exists after join modal', async function (assert) {
-    assert.expect(11);
+    assert.expect(6);
```
Comment (harshith-venkatesh) [inline tests/acceptance/live-test.js:9]: Could you explain more about change in this test case

File: tests/acceptance/live-test.js [hunk 5]
```diff
@@ -29,14 +29,5 @@ module('Acceptance | live', function (hooks) {
 
     await typeIn('[data-test-input-field]', 'sanket');
     await click('[data-test-button="live-join"]');
```
Comment (harshith-venkatesh) [inline tests/acceptance/live-test.js:31]: I see series of test cases removed after the clock of event, any particular reason for the same

File: config/environment.js [hunk 6]
```diff
@@ -43,9 +40,6 @@ module.exports = function (environment) {
   if (environment === 'test') {
     // Testem prefers this...
     ENV.locationType = 'none';
```
Comment (harshith-venkatesh) [inline config/environment.js:42]: The below credentials could be mishandled, ensure it doesn't get merged in general

File: app/services/live.js (hunk 3 above)
Comment (SanketDhabarde) [reply app/services/live.js:110]: The host needs to share the invite link with the guests to join the live share. And we need the `roomId` to join. For now, we are getting `roomId` from the log. But I am fixing it in this issue #500 point 3.

File: config/environment.js (hunk 6 above)
Comment (SanketDhabarde) [reply config/environment.js:42]: I have removed this credential from this PR. It was needed before to make the 100ms APIs work. Now we are using our own backend API so, it is not needed. For security purposes, I have disabled the room ID.

File: tests/acceptance/live-test.js (hunk 5 above)
Comment (SanketDhabarde) [reply tests/acceptance/live-test.js:31]: After the clock, the backend API call is going. Which makes the further assertion fails. We have added TODO to add the test cases for the Service.

File: tests/acceptance/live-test.js (hunk 4 above)
Comment (SanketDhabarde) [reply tests/acceptance/live-test.js:9]: https://github.com/Real-Dev-Squad/website-www/pull/498#discussion_r1240623617

File: tests/acceptance/live-test.js (hunk 5 above)
Comment (harshith-venkatesh) [reply tests/acceptance/live-test.js:31]: Should we create a ticket to address this?

**Review by harshith-venkatesh — APPROVED** [review]

File: tests/acceptance/live-test.js (hunk 5 above)
Comment (SanketDhabarde) [reply tests/acceptance/live-test.js:31]: Yes sure I will create it. Thank you.

**Review by rohan09-raj — APPROVED** [review]

**Pull request merged**